File: packages/types/product.ts
export type ProductCategory = "Pistol" | "Rifle";
export type ProductStatus = "available" | "discontinued" | "pre_order";

export interface Product {
  id: string;
  title: string;
  sku: string;
  description: string;
  price: number;
  image_url: string;
  category: ProductCategory;
  status: ProductStatus;
  weight_grams?: number;
  material?: string;
  created_at: string;
}

export interface ProductListItem {
  id: string;
  title: string;
  sku: string;
  price: number;
  image_url: string;
  category: ProductCategory;
  status: ProductStatus;
}

export interface ProductCustomization {
  color: string;
  material: string;
  engraving?: string;
}

File: apps/api/app/models/product.py
from __future__ import annotations

from decimal import Decimal
from typing import Optional

from sqlmodel import Field

from app.models.base import TimestampedBase


class Product(TimestampedBase, table=True):
    __tablename__ = "products"

    title: str = Field(index=True)
    sku: str = Field(unique=True, index=True)
    description: str = Field(default="")
    price: Decimal = Field(decimal_places=2, max_digits=10)
    image_url: str = Field(default="")
    category: str = Field(index=True)  # "Pistol" | "Rifle"
    # "available" | "discontinued" | "pre_order"
    status: str = Field(default="available", index=True)
    weight_grams: Optional[float] = Field(default=None)
    material: Optional[str] = Field(default=None)
